src/hint_processor/hint_processor_utils.rs

```diff
@@ -71,37 +67,38 @@ pub fn compute_addr_from_reference(
     //ApTracking of the Hint itself
     hint_ap_tracking: &ApTracking,
 ) -> Result<Relocatable, VirtualMachineError> {
-    let base_addr = match hint_reference.register {
-        //This should never fail
-        Some(Register::FP) => vm.get_fp(),
-        Some(Register::AP) => {
-            let var_ap_trackig = hint_reference
-                .ap_tracking_data
-                .as_ref()
-                .ok_or(VirtualMachineError::NoneApTrackingData)?;
-
-            apply_ap_tracking_correction(&vm.get_ap(), var_ap_trackig, hint_ap_tracking)?
-        }
-        None => return Err(VirtualMachineError::NoRegisterInReference),
-    };
-    if hint_reference.offset1.is_negative()
-        && base_addr.offset < hint_reference.offset1.abs() as usize
-    {
-        return Err(VirtualMachineError::FailedToGetIds);
-    }
-    if !hint_reference.inner_dereference {
-        Ok(base_addr + hint_reference.offset1 + hint_reference.offset2)
+    let offset1 = if let OffsetValue::Reference(register, offset, deref) = &hint_reference.offset1 {
+        get_offset_value_reference(
+            vm,
+            hint_reference,
+            hint_ap_tracking,
+            register,
+            offset,
+            deref,
+        )?
+        .get_relocatable()?
+        .clone()
     } else {
-        let addr = base_addr + hint_reference.offset1;
-        let dereferenced_addr = vm
-            .get_relocatable(&addr)
-            .map_err(|_| VirtualMachineError::FailedToGetIds)?;
-        let dereferenced_addr = dereferenced_addr.as_ref();
-        if let Some(imm) = &hint_reference.immediate {
-            Ok(dereferenced_addr + bigint_to_usize(imm)?)
-        } else {
-            Ok(dereferenced_addr + hint_reference.offset2)
+        return Err(VirtualMachineError::NoRegisterInReference);
+    };
+
+    match &hint_reference.offset2 {
+        OffsetValue::Reference(register, offset, deref) => {
+            // Cant add two relocatable values
+            // So OffSet2 must be Bigint
```

Comment:
I'm not sure I understand this.

Reply:
From what I read in the Python VM implementation, it is not possible to add two Relocatable values.
So, If the first term is a relocatable, the second term must be a BigInt